lib/Backend/Lower.cpp

```diff
@@ -10328,6 +10329,13 @@ Lowerer::HasSideEffects(IR::Instr *instr)
     return instr->HasAnySideEffects();
 }
 
+
+bool Lowerer::IsArgSaveRequired(Func *func) {
+    return (!func->IsTrueLeaf() || func->IsJitInDebugMode() ||
+        func->GetJnFunction()->GetIsAsmjsMode() || func->IsGeneratorFunc() || func->IsLambda() ||
```

Comment:
>GetIsAsmjsMode [](start = 31, length = 14)

Can't remember why this is needed for asm.js, but please add a comment to explain.  Do the same for generators/lambda/HasThrow as well.